query.go

```diff
@@ -2969,6 +2977,14 @@ func (q *Query) Finalize() error {
 	return nil
 }
 
+//export tiledb_go_query_submit_async_callback
+func tiledb_go_query_submit_async_callback(arg C.uintptr_t) {
+	handle := cgo.Handle(arg)
+	cont := handle.Value().(chan int)
+	handle.Delete()
+	cont <- 0
```

Comment:
[The documentation of `tiledb_query_submit_async`](https://tiledb-inc-tiledb.readthedocs-hosted.com/en/latest/c-api.html#c.tiledb_query_submit_async) says that "long-running callbacks should be dispatched to another thread". Is there a way to switch to a different thread with with certainty? Setting the channel in a new goroutine perhaps?